nautobot/extras/utils.py

```diff
@@ -409,42 +410,53 @@ def refresh_job_model_from_job_class(job_model_class, job_class):
             job_name,
         )
 
-    job_model, created = job_model_class.objects.get_or_create(
-        module_name=job_class.__module__[:JOB_MAX_NAME_LENGTH],
-        job_class_name=job_class.__name__[:JOB_MAX_NAME_LENGTH],
-        defaults={
-            "grouping": job_class.grouping[:JOB_MAX_GROUPING_LENGTH],
-            "name": job_name,
-            "is_job_hook_receiver": issubclass(job_class, JobHookReceiver),
-            "is_job_button_receiver": issubclass(job_class, JobButtonReceiver),
-            "read_only": job_class.read_only,
-            "supports_dryrun": job_class.supports_dryrun,
-            "installed": True,
-            "enabled": False,
-        },
-    )
+    try:
+        with transaction.atomic():
+            job_model, created = job_model_class.objects.get_or_create(
+                module_name=job_class.__module__[:JOB_MAX_NAME_LENGTH],
+                job_class_name=job_class.__name__[:JOB_MAX_NAME_LENGTH],
+                defaults={
+                    "grouping": job_class.grouping[:JOB_MAX_GROUPING_LENGTH],
+                    "name": job_name,
+                    "is_job_hook_receiver": issubclass(job_class, JobHookReceiver),
+                    "is_job_button_receiver": issubclass(job_class, JobButtonReceiver),
+                    "read_only": job_class.read_only,
+                    "supports_dryrun": job_class.supports_dryrun,
+                    "installed": True,
+                    "enabled": False,
+                },
+            )
+
+            if job_name != default_job_name:
+                job_model.name_override = True
```

Comment:
should also probably set `grouping_override` if `job_class.grouping` was truncated above?

Reply:
on reflection perhaps not - we don't have the same concept of renumbering a grouping in case of conflict.